lua/lspconfig/ui/lspinfo.lua

```diff
@@ -64,9 +64,18 @@ local function make_config_info(config, bufnr)
   local buffer_dir = api.nvim_buf_call(bufnr, function()
     return vim.fn.expand '%:p:h'
   end)
-  local root_dir = config.get_root_dir and config.get_root_dir(buffer_dir)
-  if root_dir then
-    config_info.root_dir = root_dir
+
+  if config.get_root_dir then
+    local root_dir
+    local status = coroutine.resume(coroutine.create(function()
+      local status, err = pcall(function()
+        root_dir = config.get_root_dir(buffer_dir)
+      end)
+      if not status then
+        print('lspconfig: unhandled error: ' .. tostring(err))
+      end
+    end))
+    config_info.root_dir = status and root_dir or error_messages.root_dir_not_found
```

Comment:
If we don't support async root_dir functions right now then there should be at least an error message that says that
```suggestion
    local co = coroutine.create(function()
      local status, err = pcall(function()
        root_dir = config.get_root_dir(buffer_dir)
      end)
      if not status then
        print('lspconfig: unhandled error: ' .. tostring(err))
      end
    end)
    coroutine.resume(co)
    if root_dir then
      config_info.root_dir = root_dir
    elseif coroutine.status(co) == 'suspended' then
      config_info.root_dir = error_messages.async_root_dir_function
    else
      config_info.root_dir = error_messages.root_dir_not_found
    end
```
And add to `error_messages`:
```lua
  async_root_dir_function = 'Asynchronous root_dir functions are not supported in :LspInfo',
```

Reply:
Actually, `root_dir` in rust looks so simple that it should be very easy to support both synchronous and asynchronous versions. If `coroutine.running()` is `nil` then you just call `vim.fn.jobwait`. Maybe for now `root_dir` functions should be required to support synchronous version just to keep `:LspInfo` working